compiler/p/codegen/FPTreeEvaluator.cpp

```diff
@@ -2016,159 +2015,6 @@ TR::Register *OMR::Power::TreeEvaluator::ifdcmpleuEvaluator(TR::Node *node, TR::
    return NULL;
    }
 
-// also handles TR::fcmpeq
-TR::Register *OMR::Power::TreeEvaluator::dcmpeqEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-   {
-   return compareFloatAndSetOrderedBoolean(TR::InstOpCode::beq, TR::InstOpCode::bad, 0, node, cg);
-   }
-
-// also handles TR::fcmpne
-TR::Register *OMR::Power::TreeEvaluator::dcmpneEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-   {
-   return compareFloatAndSetOrderedBoolean(TR::InstOpCode::bne, TR::InstOpCode::bad, 0, node, cg);
-   }
-
-// also handles TR::fcmplt
-TR::Register *OMR::Power::TreeEvaluator::dcmpltEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-   {
-   int64_t imm = 0;
-   if (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P9))
```

Comment:
Did we lose this P9 optimization (e.g. expanding setbool to setb) ?

Reply:
Nice catch! I completely missed that this optimization originally existed, as it was only applied to `[fd]cmplt` and `[fd]cmpgt` and its implementation was hidden amongst code in the control flow instruction expansion despite not requiring any control flow.

AFAICT there's no reason not to also apply this same optimization to other compares as well to make all compares branchless on P9. It's obviously not as nice as P10 (it requires some extra instructions to set things up correctly), but it should still work. I've gone ahead and added exploitation of `setb` across the board for all compares in 1658bdf2c4ac5d5a509b0f7abe59b4b793cb621d.